RecoLocalCalo/HcalRecAlgos/src/parseHBHEPhase1AlgoDescription.cc

```diff
@@ -76,21 +106,28 @@ parseHBHEPhase1AlgoDescription(const edm::ParameterSet& ps)
 
     if (className == "SimpleHBHEPhase1Algo")
     {
-        std::unique_ptr<PulseShapeFitOOTPileupCorrection> m2;
-        if (ps.getParameter<bool>("useM2"))
-            m2 = parseHBHEMethod2Description(ps);
-
+	std::unique_ptr<MahiFit> mahi;
+	std::unique_ptr<PulseShapeFitOOTPileupCorrection> m2;
         std::unique_ptr<HcalDeterministicFit> detFit;
-        if (ps.getParameter<bool>("useM3"))
-            detFit = parseHBHEMethod3Description(ps);
+
+	// only run Mahi OR (Method 2 & Method 3) but not both
+	if (ps.getParameter<bool>("useMahi"))
+	  mahi = parseHBHEMahiDescription(ps);
+	else {
+	  if (ps.getParameter<bool>("useM2"))
+	    m2 = parseHBHEMethod2Description(ps);
+	  if (ps.getParameter<bool>("useM3"))
+	    detFit = parseHBHEMethod3Description(ps);
+	}
```

Comment:
please add a consistency check so that both useMahi and useM2/M3 are not set at configuration

Reply:
I'm not sure how to technically implement this, sorry. Could you clarify?

Reply:
```if (useMahi && (useM3 || useM2)) throw cms::Exception("ConfigurationError")<<"such and such algo does not accept simultaneous configuration of parameters xyz"```